StockTrader/src/main/java/com/restResource/StockTrader/controller/AddController.java

```diff
@@ -24,24 +38,19 @@ public AddController(AccountRepository accountRepository, LoggingService logging
     HttpStatus addToAccountBalance(@RequestParam String userId,
                                    @RequestParam int amount) {
 
-        loggingService.logUserCommand(
-                UserCommandLog.builder()
-                        .username(userId)
-                        .funds(amount)
-                        .command(CommandType.ADD)
-                        .build());
-
+        loggingService.logUserCommand(CommandType.ADD, userId, null, null, amount);
+        loggingService.logDebugEvent(CommandType.ADD,userId,null,null,amount,"Trying to add funds to this guys account");
         try {
-            if( amount <= 0 ) {
+            if (amount <= 0) {
                 throw new IllegalArgumentException(
                         "The ADD amount parameter must be greater than zero");
-            }
-            else {
+            } else {
                 accountRepository.updateAccountBalance(userId, amount);
             }
             return HttpStatus.OK;
-        } catch( IllegalArgumentException e ) {
+        } catch (Exception e) {
             System.out.println("Exception in AddController: " + e.toString());
+            loggingService.logErrorEvent(CommandType.ADD,userId,null,null,amount,"Amount added must be lteq 0");
```

Comment:
This would be a great place to use a builder type. Would avoid the multiple null params, and would be much more readable and less error prone (easy to get the order mixed up here). It's fine for now, we have a deadline, but I'd like it changed in the future.

Reply:
Sure. I tried to continue to use builder but I was getting null object types. But yeah you're right, builder form is nicer.